BrowserKit/Sources/WebEngine/WKWebview/WKEngineConfigurationProvider.swift

```diff
@@ -13,41 +13,63 @@ public struct WKWebviewParameters {
+    var schemeHandler: SchemeHandler
+
+    public init(blockPopups: Bool,
+                isPrivate: Bool,
+                autoPlay: WKAudiovisualMediaTypes,
+                schemeHandler: SchemeHandler) {
         self.blockPopups = blockPopups
         self.isPrivate = isPrivate
+        self.autoPlay = autoPlay
+        self.schemeHandler = schemeHandler
     }
 }
 
 /// Provider to get a configured `WKEngineConfiguration`
-protocol WKEngineConfigurationProvider {
-    func createConfiguration() -> WKEngineConfiguration
+/// Only one configuration provider per window should exists in the application.
+public protocol WKEngineConfigurationProvider {
+    func createConfiguration(parameters: WKWebviewParameters) -> WKEngineConfiguration
 }
 
-struct DefaultWKEngineConfigurationProvider: WKEngineConfigurationProvider {
-    private let parameters: WKWebviewParameters
+/// FXIOS-TODO - This will be internal when the WebEngine is fully integrated in Firefox iOS
+public struct DefaultWKEngineConfigurationProvider: WKEngineConfigurationProvider {
+    private static let normalSessionsProcessPool = WKProcessPool()
+    private static let privateSessionsProcessPool = WKProcessPool()
 
-    init(parameters: WKWebviewParameters) {
-        self.parameters = parameters
-    }
+    public init() {}
 
-    func createConfiguration() -> WKEngineConfiguration {
+    public func createConfiguration(parameters: WKWebviewParameters) -> WKEngineConfiguration {
         let configuration = WKWebViewConfiguration()
-        configuration.processPool = WKProcessPool()
+
+        // Since our app creates multiple web views, we assign the same WKProcessPool object to web views that
+        // may safely share a process space
+        configuration.processPool = parameters.isPrivate
+        ? DefaultWKEngineConfigurationProvider.privateSessionsProcessPool
+        : DefaultWKEngineConfigurationProvider.normalSessionsProcessPool
+
         configuration.preferences.javaScriptCanOpenWindowsAutomatically = !parameters.blockPopups
+        configuration.mediaTypesRequiringUserActionForPlayback = parameters.autoPlay
         configuration.userContentController = WKUserContentController()
         configuration.allowsInlineMediaPlayback = true
+
+        // TODO: FXIOS-8086 - Evaluate if ignoresViewportScaleLimits is still needed
+        // We do this to go against the configuration of the <meta name="viewport">
+        // tag to behave the same way as Safari :-(
+        configuration.ignoresViewportScaleLimits = true
         if #available(iOS 15.4, *) {
             configuration.preferences.isElementFullscreenEnabled = true
         }
-        if parameters.isPrivate {
-            configuration.websiteDataStore = WKWebsiteDataStore.nonPersistent()
-        } else {
-            configuration.websiteDataStore = WKWebsiteDataStore.default()
-        }
+        configuration.websiteDataStore = parameters.isPrivate
```

Comment:
We need to be careful here. Calling this function multiple times can lead to issues like [this bug](https://mozilla-hub.atlassian.net/browse/FXIOS-11833). We can't pinpoint why exactly this happens, but just calling to `WKWebsiteDataStore.nonPersistent()` more than needed has caused us cookie loss in the past.

Reply:
I'll add a comment, but we should have only `WKEngineConfigurationProvider` per app (or per window?)